src/queue.c

```diff
@@ -4498,20 +4550,36 @@ _dispatch_get_main_queue_port_4CF(void)
 	return (mach_port_t)dq->do_ctxt;
 }
 
-static bool main_q_is_draining;
+void
+_dispatch_main_queue_callback_4CF(mach_msg_header_t *msg DISPATCH_UNUSED)
+{
+	if (main_q_is_draining) {
+		return;
+	}
+	_dispatch_queue_set_mainq_drain_state(true);
+	_dispatch_main_queue_drain();
+	_dispatch_queue_set_mainq_drain_state(false);
+}
 
-// 6618342 Contact the team that owns the Instrument DTrace probe before
```

Comment:
not ok to remove a comment that says something quite explicit about it not being removed ever.